src/Hub.cpp

```diff
@@ -289,7 +315,49 @@ const sc2::Point3D* Hub::GetNextExpansion() {
         return nullptr;
 
     it->owner = Owner::CONTESTED;
-    return &(it->town_hall_location);
+    return &(*it);
+}
+
+Expansion* Hub::GetExpansionOfTownhall(const sc2::Unit& unit_) {
+    for (auto& i : m_expansions) {
+        if (std::floor(i.town_hall_location.x) == std::floor(unit_.pos.x) &&
+            std::floor(i.town_hall_location.y) == std::floor(unit_.pos.y)) {
+            return &i;
```

Comment:
Looks dangerous as a pointer to reference returned and the reference itself is local variable. Here you should return something from _expansions.

Reply:
I've looked around the internet, and seen some say that references do not have a memory address location, so &i is returning the address to the actual Expansion object, not a local variable, as i is not an iterator in this case.

Otherwise, I should be able to get the memory address inside the reference in order to create an Expansion*, but I cannot, because it already refers directly to the object.

Reply:
Thats why I said that it looks risky. But the 100% safe would be to use traditional for loop here which I think is the best.

Reply:
I overloaded the Expansion & address-of operator and confirmed that &i is calling the address-of of the Expansion object itself, not the reference's address.

So, I can confirm that &i is 100% safe, also.

This is included in the C++ Standard:
https://isocpp.org/wiki/faq/references
"This includes taking the address of it. The values of &i and &x are identical."
```
void swap(int& i, int& j)
{
  int tmp = i;
  i = j;
  j = tmp;
}
int main()
{
  int x, y;
  swap(x,y);
}
```

Reply:
https://isocpp.org/wiki/faq/references is the official C++ Standard site, and says the &object and its &reference are identical.
"Important note: Even though a reference is often implemented using an address in the underlying assembly language, please do not think of a reference as a funny looking pointer to an object. A reference is the object, just with another name. It is neither a pointer to the object, nor a copy of the object. It is the object. There is no C++ syntax that lets you operate on the reference itself separate from the object to which it refers."

Again: "There is no C++ syntax that lets you operate on the reference itself separate from the object to which it refers."
Which means it is impossible to get the address of a reference variable in standard C++.

If there are non-standard-compliant compilers we want to support, -- or that the purpose of not using it is so that new coders won't accidentally try to use the same style to return the address of local iterators, then I completely understand. I defer to your expertise on platforms/compilers.

I prefer the simplicity of this style, but I don't want to hold up the PR for this.

Can this stay in, or do you really want me to change it to a For loop? I trust your decision. Спасибо

Reply:
I have refactored this to use an iterator and then an explicit &(*it) as is used elsewhere already.